code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -424,4 +433,30 @@ GLOBAL_LIST_EMPTY(ts_spiderling_list)
 				var/pc_of_max_per_second = round(((hp_points_per_second / maxHealth) * 100), 0.1)
 				stat(null, "Regeneration: [ltext]: <font color='[lcolor]'>[num2text(pc_of_max_per_second)]% of health per second</font>")
 
-
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoRemoteView()
+	if(client && (client.eye != client.mob))
+		reset_perspective()
+		return
+	if(health != maxHealth)
+		to_chat(src, "<span class='warning'>You must be at full health to do this!</span>")
+		return
+	var/list/targets = list()
+	targets += src // ensures that self is always at top of the list
+	for(var/thing in GLOB.ts_spiderlist)
+		var/mob/living/simple_animal/hostile/poison/terror_spider/T = thing
+		if(T.stat == DEAD)
+			continue
+		if(T.spider_awaymission != spider_awaymission)
+			continue
+		if(T == src)
+			continue
+		targets += T
+	var/mob/living/L = input("Choose a terror to watch.", "Selection") in targets
```

Comment:
I know I'm very spoiled by the goodies we have, but it'd be so cool if this was TGUI !

Reply:
Cool would be combining this and the 'hive sense' into one ability which gives you a window listing all spiders with a 'follow' button for each. Unfortunately, that is not in scope right now. This is still a draft and I'm not even sure this will be final design. So I don't really want to start re-coding the hivesense ability and TGUI-ing it all in this PR. I might in a subsequent PR though once this has been live for a bit and I'm sure the balance is fine.